From 9382b6a75fb1e7bf6c73d1185e8119f8bb34c563 Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Thu, 23 Feb 2017 17:49:45 -0600
Subject: [PATCH] CMSettingsProvider: remove per-user cursor observation

We don't need it because we don't support using the returned cursor
directly, and it can cause deadlocks when being called from platform
code.

Change-Id: I2f85be1152569ba27e4622d310d867e20965faa3
---
 .../src/org/cyanogenmod/cmsettings/CMSettingsProvider.java    | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/packages/CMSettingsProvider/src/org/cyanogenmod/cmsettings/CMSettingsProvider.java b/packages/CMSettingsProvider/src/org/cyanogenmod/cmsettings/CMSettingsProvider.java
index d730584..a4630df 100644
--- a/packages/CMSettingsProvider/src/org/cyanogenmod/cmsettings/CMSettingsProvider.java
+++ b/packages/CMSettingsProvider/src/org/cyanogenmod/cmsettings/CMSettingsProvider.java
@@ -470,17 +470,6 @@ private Cursor queryForUser(int userId, Uri uri, String[] projection, String sel
                     null, sortOrder);
         }
 
-        // the default Cursor interface does not support per-user observation
-        try {
-            AbstractCursor abstractCursor = (AbstractCursor) returnCursor;
-            abstractCursor.setNotificationUri(getContext().getContentResolver(), uri, userId);
-        } catch (ClassCastException e) {
-            // details of the concrete Cursor implementation have changed and this code has
-            // not been updated to match -- complain and fail hard.
-            Log.wtf(TAG, "Incompatible cursor derivation");
-            throw e;
-        }
-
         return returnCursor;
     }
 
